src/internal_photoev.py

```diff
@@ -129,6 +129,12 @@ def __call__(self, disc, dt):
         sigmadot = self.Sigmadot(disc)
         Sigma_new = disc.Sigma - dt * sigmadot
 
+        # Check that the surface density never becomes negative
+
+        for i in range(0, len(Sigma_new)-1):
```

Comment:
Try to avoid for loops - it makes the code very slow. you can do this in a vectorised way (e.g. Sigma_new[Sigma_new<0]=self._floor_density